splinterd/src/config/toml.rs

```diff
@@ -16,6 +16,8 @@
 
 use crate::config::PartialConfigBuilder;
 use crate::config::{ConfigError, ConfigSource, PartialConfig};
+#[cfg(feature = "log-config")]
+use log4rs::config::RawConfig as LogConfig;
```

Comment:
Remove use of log4rs from within crate::config and define the appropriate structs explicitly. This is important for two reasons: 1) log4rs is fundamentally unrelated to configuration loading; and 2) the config file is a versioned API and changes to it need to be handled in a backward-compatible way -- so we can not introduce an external data structure as part of our API definition.

Then, write an Into trait allowing conversion of our LogConfig struct to log4rs's RawConfig. This should maybe go in a crate::log module.

Reply:
fixed/updated